include/fw_info.h

```diff
@@ -77,28 +77,30 @@ OFFSET_CHECK(struct fw_info_ext_api, ext_api_version, 24);
  */
 
 
-/* Decorator for struct fw_info_ext_api instances to place them in the correct
+/* Macro for initializing struct fw_info_ext_api instances in the correct
  * linker section. Also creates a u8_t in another section to provide a count of
  * the number of struct fw_info_ext_api instances.
  */
-#define EXT_API(type, name) \
+#define EXT_API(ext_api_name, type, name) \
 	Z_GENERIC_SECTION(.ext_apis) \
 	const u8_t _CONCAT(name, _ext_api_counter) = 0xFF; \
 	BUILD_ASSERT_MSG((sizeof(type) % 4) == 0, \
-			"EXT_API " #type " is not word-aligned"); \
-	Z_GENERIC_SECTION(.firmware_info.1) __attribute__((used)) \
-	const type name
-
-
-/* Macro for initializing struct fw_info_ext_api instances. */
-#define FW_INFO_EXT_API_INIT(id, flags, version, total_size) \
+			"Size of EXT_API " #type " is not word-aligned"); \
+	struct __packed _CONCAT(name, _t) \
 	{ \
-		.magic = {EXT_API_MAGIC},\
-		.ext_api_len = total_size, \
-		.ext_api_id = id, \
-		.ext_api_flags = flags, \
-		.ext_api_version = version, \
-	}
+		struct fw_info_ext_api header; \
+		type ext_api; \
+	}; \
+	Z_GENERIC_SECTION(.firmware_info.1) __attribute__((used)) \
```

Comment:
is numbers used for ordering?

Reply:
Yes. The list contains provided APIs first, then API requests.